# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/DrawTargetWebgl.h
# Commit: 7fea32057b92
# Full Hash: 7fea32057b920eac49960541efa2f3fea9074c19
# Author: Lee Salzman <lsalzman@mozilla.com>
# Date: 2022-02-12 09:47:43
# Regressor Bug: 1754130
# File Overlap Count: 2
# Description:
#   Bug 1754130 - Support presenting a WebGLFramebuffer to its own swap chain without opaque FB. r=aosmond,jgilbert
#   
#   Most of the support for presenting a WebGLFramebuffer to a swap chain existed as part of the
#   mechanism for opaque WebXR framebuffer support. However, such "opaque" framebuffer are meant
#   to be opaque in the sense that their attachments can't be inspected or changed, which does
# ==============================================================================

diff -r 1ac8f10e0025 -r 7fea32057b92 dom/canvas/DrawTargetWebgl.h
--- a/dom/canvas/DrawTargetWebgl.h	Fri Feb 11 15:36:30 2022 +0000
+++ b/dom/canvas/DrawTargetWebgl.h	Fri Feb 11 15:36:30 2022 +0000
@@ -205,7 +205,7 @@
     bool FillGlyphsAccel(ScaledFont* aFont, const GlyphBuffer& aBuffer,
                          const Pattern& aPattern, const DrawOptions& aOptions);
 
-    void PruneTextureHandle(RefPtr<TextureHandle> aHandle);
+    void PruneTextureHandle(const RefPtr<TextureHandle>& aHandle);
     bool PruneTextureMemory(size_t aMargin = 0, bool aPruneUnused = true);
 
     bool RemoveSharedTexture(const RefPtr<SharedTexture>& aTexture);
@@ -332,6 +332,10 @@
   void SetTransform(const Matrix& aTransform) override;
   void* GetNativeSurface(NativeSurfaceType aType) override;
 
+  Maybe<layers::SurfaceDescriptor> GetFrontBuffer();
+
+  bool CopySnapshotTo(DrawTarget* aDT);
+
   operator std::string() const {
     std::stringstream stream;
     stream << "DrawTargetWebgl(" << this << ")";